File: backend/app/model/dto/request/updateProfileDto.py
from pydantic import BaseModel
from typing import Optional


class UpdateProfileDto(BaseModel):
    display_name: Optional[str] = None
    preferred_email: Optional[str] = None
    phone: Optional[str] = None
    sex: Optional[str] = None
    birth_year: Optional[int] = None
    daily_step_goal: Optional[int] = None
    timezone: Optional[str] = None
    height: Optional[float] = None  # in cm
    weight: Optional[float] = None  # in kg

File: mobile/src/main/java/com/example/harbit/data/remote/dto/UpdateProfileDto.kt
package com.example.harbit.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UpdateProfileDto(
    @SerialName("display_name")
    val displayName: String? = null,
    
    @SerialName("preferred_email")
    val preferredEmail: String? = null,
    
    @SerialName("phone")
    val phone: String? = null,
    
    @SerialName("sex")
    val sex: String? = null,
    
    @SerialName("birth_year")
    val birthYear: Int? = null,
    
    @SerialName("daily_step_goal")
    val dailyStepGoal: Int? = null,
    
    @SerialName("timezone")
    val timezone: String? = null,
    
    @SerialName("height")
    val height: Float? = null,
    
    @SerialName("weight")
    val weight: Float? = null
)
